### PR TITLE
Update react 18 antd5

### PR DESCRIPTION
**Pull request by kestarumper**
Based on the `update-react` branch. Update React to v18 in the `uniforms-antd` package. Updating AntD to v5 is required.

### CODE REVIEW

File: packages/uniforms-antd/package.json [hunk 1]
```diff
@@ -32,8 +32,10 @@
     "src/*.tsx"
   ],
   "peerDependencies": {
-    "@ant-design/icons": "^4.0.0",
-    "antd": "^4.0.0",
+    "@ant-design/icons": "^5.0.0",
+    "antd": "^5.0.0",
+    "dayjs": "^1.11.13",
+    "rc-input-number": "^9.2.0",
```
Comment (kestarumper) [inline packages/uniforms-antd/package.json:38]: Fixes
```
The inferred type of 'default' cannot be named without a reference to '.pnpm/rc-input-number@9.2.0_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/rc-input-number'. This is likely not portable. A type annotation is necessary.ts(2742)
```

Comment (piotrpospiech) [reply packages/uniforms-antd/package.json:38]: Let's do this differently. 

1. Keep `rc-input-number@9.2.0` only in `devDependencies`
2. Add `/// <reference types="rc-input-number" />` to the `NumField` file

File: packages/uniforms-antd/package.json [hunk 2]
```diff
@@ -32,8 +32,10 @@
     "src/*.tsx"
   ],
   "peerDependencies": {
-    "@ant-design/icons": "^4.0.0",
-    "antd": "^4.0.0",
+    "@ant-design/icons": "^5.0.0",
+    "antd": "^5.0.0",
+    "dayjs": "^1.11.13",
```
Comment (piotrpospiech) [inline packages/uniforms-antd/package.json:37]: I think we can consider keeping only `antd` and `react` in `peerDependencies` for now.

**Review by piotrpospiech — CHANGES_REQUESTED** [review]
1. AntD tests are skipped. You need to remove `.skip` from them. 5 AntD tests fail. 
2. By the way, our CI fails, so we need to adjust other tests as well. EDIT: This is caused by using `npm` by CI. Let's fix that at the end.

**Review by piotrpospiech — APPROVED** [review]

**Pull request merged**